website/docs/cookbooks/scala-jvm.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Test your code with java 8 
+sidebar_position: 4
+---
+
+You can use Scala CLI to test your code compatibility with various versions of `java` (no manual installation of java is required).It automatically downloads the specified version of java.
+
+The following snippet used the new method `Files.writeString` from `java` 11. 
+
+```scala name:Main.scala
+import java.nio.file.Files
+import java.nio.file.Paths
+
+object Main extends App {
+    val filePath = Files.writeString(
+      Files.createTempFile(
+        Paths.get("").toAbsolutePath(),
+        "scala-cli",
+        ".txt"
+      ),
+      "Hello from ScalaCli"
+    )
+    val fileContent: String = Files.readString(filePath)
+  }
+ ```
+
+
+Passing `--jvm` to the `scala-cli` command and run your application with the specified java version.
+
+```scala-cli 
+scala-cli Main.scala --jvm adopt:11
+```
+
+<!-- Expected:
+-->
+
+To test your application with java 8, change the value of `--jvm` parameter.
+```bash
+scala-cli Main.scala  --jvm adopt:8
+# In this case, it raises an error because the `Files.createTempFile` method is not available in java 8
+#
+# Exception in thread main: java.lang.Exception: java.lang.NoSuchMethodError: java.nio.file.Files.writeString(Ljava/nio/file/Path;Ljava/lang/CharSequence;[Ljava/nio/file/OpenOption;)Ljava/nio/file/Path;
```

Comment:
@alexarchambault isn't it bad behaviour? Should code simply does not compile?

Reply:
I specially prepared a code snippet that shouldn't compile in jvm 8.

Reply:
JVM switching only works well at runtime, and less so for compiling. When compiling, the JVM of Bloop is actually used.